fastn-core/src/library2022/processor/sqlite.rs

```diff
@@ -176,64 +200,86 @@ fn extract_named_parameters(
-                    .map(|v| Box::new(v) as Box<dyn rusqlite::ToSql>);
-
-            match param_value {
-                Ok(v) => params.push(v),
-                Err(_) => {
-                    // todo: Handle Error
+            State::InsideEscapeSequence(escape_count) => {
+                if c == '\\' {
+                    state = State::InsideEscapeSequence(escape_count + 1);
+                } else {
+                    state = State::InsideStringLiteral;
                 }
             }
+            State::StartTypeHint => {
+                if c == ':' {
+                    state = State::InsideTypeHint;
+                } else {
+                    state = State::ParseError("Type hint must start with `::`".to_string());
+                }
+            }
+            State::InsideParam => {
+                if c == ':' {
+                    state = State::StartTypeHint;
+                } else if c.is_alphanumeric() {
+                    param_name.push(c);
+                } else if c == ',' || c == ';' || c.is_whitespace() && !param_name.is_empty() {
+                    state = State::PushParam;
+                }
+            }
+            State::InsideTypeHint => {
+                if c.is_alphanumeric() {
+                    param_type.push(c);
+                } else {
+                    state = State::PushParam;
+                }
+            }
+            State::PushParam => {
+                state = State::OutsideParam;
 
-            // clear param_name and param_type
-            param_name.clear();
-            param_type.clear();
-        }
-    }
+                let param_value =
+                    resolve_param(&param_name, &param_type, doc, &headers, line_number)?;
 
-    // handle the last param if there was no trailing comma or space
-    if !param_name.is_empty() {
-        let param_value =
-            resolve_variable_from_headers(&param_name, &param_type, doc, &headers, line_number)
-                .or_else(|_| resolve_variable_from_doc(&param_name, doc, line_number))
-                .map(|v| Box::new(v) as Box<dyn rusqlite::ToSql>);
+                params.push(Box::new(param_value) as Box<dyn rusqlite::ToSql>);
 
-        match param_value {
-            Ok(v) => params.push(v),
-            Err(_) => {
-                // todo: handle the error
+                param_name.clear();
+                param_type.clear();
+            }
+            State::ParseError(error) => {
+                return Err(ftd::interpreter::Error::ParseError {
+                    message: format!("Failed to parse SQL Query: {}", error),
+                    doc_id: doc.name.to_string(),
+                    line_number,
+                });
             }
         }
     }
 
+    // Handle the last param if there was no trailing comma or space
+    if let State::PushParam = state {
+        println!("Here");
+        if !param_name.is_empty() {
```

Comment:
```suggestion
    if state.eq(State::PushParam) && !param_name.is_empty() {
```

Reply:
Error: "Method 'eq' is not found for this enum," but can we use 'matches!' instead?

Reply:
Eg: `if matches!(state, State::PushParam) && !param_name.is_empty() {`

Reply:
@harshdoesdev https://github.com/fastn-stack/fastn/pull/1423#discussion_r1377094773